pkg/skaffold/schema/validation/validation.go

```diff
@@ -77,6 +78,81 @@ func validateImageNames(artifacts []*latest.Artifact) (errs []error) {
 	return
 }
 
+func validateArtifactDependencies(artifacts []*latest.Artifact) (errs []error) {
+	errs = append(errs, validateArtifactDependencyUniqueness(artifacts)...)
+	errs = append(errs, validateArtifactCircularDependencies(artifacts)...)
+	return
+}
+
+// validateArtifactDependencies makes sure all artifact dependencies are found and don't have cyclic references
+func validateArtifactCircularDependencies(artifacts []*latest.Artifact) (errs []error) {
+	m := make(map[string]*latest.Artifact)
+	for _, artifact := range artifacts {
+		m[artifact.ImageName] = artifact
+	}
+	visited := make(map[string]bool)
+	for _, artifact := range artifacts {
+		if err := dfs(artifact, visited, make(map[string]bool), m); err != nil {
+			errs = append(errs, err)
+			return
+		}
+	}
+	return
+}
+
+func dfs(artifact *latest.Artifact, visited, marked map[string]bool, artifacts map[string]*latest.Artifact) error {
```

Comment:
Credit the algorithm source?

Reply:
It's just `Cycle Detection in a Directed Graph using Depth First Search`. I don't think this has any other name.

Reply:
added a description with that